vmm/src/lib.rs

```diff
@@ -457,18 +457,21 @@ impl Vmm {
         Ok(())
     }
 
-    fn vm_info(&self) -> result::Result<VmInfo, VmError> {
+    fn vm_info(&mut self) -> result::Result<VmInfo, VmError> {
         match &self.vm_config {
             Some(config) => {
                 let state = match &self.vm {
                     Some(vm) => vm.get_state()?,
                     None => VmState::Created,
                 };
 
-                Ok(VmInfo {
-                    config: Arc::clone(config),
-                    state,
-                })
+                let config = Arc::clone(config);
+
+                if let Some(ref mut vm) = self.vm {
+                    config.lock().unwrap().memory.balloon_size = vm.get_balloon_actual()?;
```

Comment:
Please don't do that, it's not the role of `vm.info` to update the configuration.
You could extend the `VmInfo` structure so that it contains a dedicated field for the balloon actual size, but don't update the configuration.

Reply:
Added memory_actual_size to vm.info to show the current memory size of vm.  This bahaviour like the qemu command "info balloon".